### PR TITLE
Add CMD and ENTRYPOINT (fixes #130)

### LINKED ISSUE #130: Support shell syntax
Right now, the CMD is specified in exec form no matter what.
This means, that you can't really use variables in you command (adding `sh -c`to the beginning doesn't always work).

It would be nice if you could choose between shell and exec form.

### PR DESCRIPTION
**Pull request by osigida**
introduced cmd and entrypoint with exec and shell form support.
the change is not backward compatible to the previous version.

Signed-off-by: Oleg Sigida <oleg.sigida@gmail.com>

### CODE REVIEW

**Comment by alan-czajkowski** [comment]
please fix the documentation: manual.md mentions entrypoint but still shows ```<cmd>``` tag

File: doc/manual.md [hunk 1]
```diff
@@ -262,7 +265,30 @@ Here's an example:
...
+    <!-- or exec form for command -->
+    <params>
+      <param>java</param>
+      <param>-jar</param>
+      <param>/opt/demo/server.jar</param>
+    </params>
+  </cmd>
+
+  <!-- or you can use entry point -->
+  <cmd>
```
Comment (alan-czajkowski) [inline doc/manual.md]: please fix the documentation, mentions entrypoint but still shows ```<cmd>``` tag

File: src/main/java/org/jolokia/docker/maven/assembly/DockerFileBuilder.java [hunk 2]
```diff
@@ -14,6 +16,8 @@
  */
 public class DockerFileBuilder {
 
+    private static final Joiner JoinOnComma = Joiner.on("\",\"");
```
Comment (alan-czajkowski) [inline src/main/java/org/jolokia/docker/maven/assembly/DockerFileBuilder.java]: should use lowercase J? Java coding conventions?

**Comment by alan-czajkowski** [comment]
can CMD and ENTRYPOINT configuration in the plugin be used in conjunction? the Docker documentation mentions that if ENTRYPOINT is used then CMD becomes the parameters to ENTRYPOINT, just want to make sure the plugin will handle both simultaneously

**Comment by osigida** [comment]
@alan-czajkowski yes, if both cmd and entrypoint are specified they will be printed out to docker file, so one an use this feature (gonna test it soon, sounds like a good simplification to my use cases)

File: doc/manual.md (hunk 1 above)
Comment (osigida) [inline doc/manual.md]: oops, thanks a lot! fixed

File: src/main/java/org/jolokia/docker/maven/assembly/DockerFileBuilder.java (hunk 2 above)
Comment (osigida) [inline src/main/java/org/jolokia/docker/maven/assembly/DockerFileBuilder.java]: actually it should be JOIN_ON_COMMA as a static constant, renamed.

**Comment by rhuss** [comment]
Thanks a lot for the PR, it adds really a very useful feature and is well done. 
However, since it will change the configuration syntax in such a way that a migration is necessary, I would like to schedule it for 0.12.0 (second version number is reserved for such things). I need to do a 0.11.4 before for some important fixes, though. But 0.12.0 is the next one ;-)

I added some commit comments, i.e. it would be perfect if you could adapt the PR a bit:

* Remove the dependency to guava, it's not (yet) really needed and only adds to the baggage.
* Adapt the example in the documentation so that it is directly usable ;-) 
* Squash commits to a single one, please. This makes commenting on commits much easier. Tipp: Check for 'git rebase -i' and search for "git squash", e.g. [this](http://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git) one)

As I said, I really like the code, its great, these are only minor things. Thanks a lot !

**Comment by rhuss** [comment]
Ah, I forgot. Could you please also rebase againt branch `integration`. This is where I integrate all PRs. Danger of conflict is quite low, though, the current PRs are quite orthogonal.

**Comment by osigida** [comment]
@rhuss, sure, will update the PR soon.
Regarding guava, I found out that guava 10 is already in the dependency list, then I decided to keep it 
So dependency wise it is the same, but it is visible now.

So what will be the conclusion, keep or drop?

**Comment by rhuss** [comment]
Ah, ok, its a transitive dep of maven-plugin-api. No problem to keep it then ;-)

**Comment by osigida** [comment]
@rhuss, please have a look
* as we agreed, guava is still used
* restructured documentation a bit, is it better?
* commits are squashed
* rebased on recent integration

**Comment by rhuss** [comment]
thanks, will review and add this for 0.12.0 (there will a quick bugfix release 0.11.5 before btw).

**Comment by osigida** [comment]
...rebased on integration branch...

**Comment by rhuss** [comment]
Thanks for the rebase. I will review it ASAP, but probably not this week anymore, hopefully next week.

**Comment by rhuss** [comment]
Merged in and tuned it a bit. Sorry for the long delay.

BTW, I renamed `<params><param>` to `<exec><arg>` since it better matches the Dockerfile syntax.

Thanks again !

**Comment by alan-czajkowski** [comment]
when will a release be made for 0.12.0?

**Comment by rhuss** [comment]
Next week, promised. Just have to prepare a talk for Thursday, sorry, i'm in 'slide machine' mode at the moment.

**Pull request merged**